File: frontend/src/types/game.ts
export type GameCategory = 'sub-13' | 'sub-15';

export interface Player {
  id: string;
  name: string;
  number: number;
}

export interface GameStats {
  id: number;
  player_id: number;
  game_id: number;
  points: number;
  rebounds: number;
  assists: number;
  steals: number;
  blocks: number;
  fouls: number;
  turnovers: number;
  minutes_played: number;
  created_at: string;
  updated_at: string;
}

export interface PlayerGameStats {
  playerId: string;
  stats: GameStats;
}

export interface Game {
  id?: number | string;
  opponent?: string;
  date?: string;
  location?: string;
  name?: string;
  time?: string;
  category?: GameCategory;
  stats?: PlayerGameStats[];
  players?: Player[];
  createdAt?: string;
  updatedAt?: string;
}

export interface GameAction {
  type: 'ADD_STAT' | 'REMOVE_STAT';
  playerId: string;
  statType: keyof GameStats;
  value: number;
  timestamp: number;
}

// Tipos para criação e atualização
export interface GameCreate {
  opponent: string;
  date: string;
  location?: string;
  category?: GameCategory;
  time?: string;
}

export interface GameUpdate {
  opponent?: string;
  date?: string;
  location?: string;
  category?: GameCategory;
  time?: string;
  status?: string;
}

export interface GameStatsCreate {
  player_id: number;
  game_id: number;
  points?: number;
  rebounds?: number;
  assists?: number;
  steals?: number;
  blocks?: number;
  fouls?: number;
  turnovers?: number;
  minutes_played?: number;
  quarter?: number;
  two_attempts?: number;
  two_made?: number;
  three_attempts?: number;
  three_made?: number;
  free_throw_attempts?: number;
  free_throw_made?: number;
  interference?: number;
}

export interface GameStatsUpdate {
  points?: number;
  rebounds?: number;
  assists?: number;
  steals?: number;
  blocks?: number;
  fouls?: number;
  turnovers?: number;
  minutes_played?: number;
  quarter?: number;
  two_attempts?: number;
  two_made?: number;
  three_attempts?: number;
  three_made?: number;
  free_throw_attempts?: number;
  free_throw_made?: number;
  interference?: number;
} 

File: backend/app/schemas/game.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime
from app.schemas.player import PlayerOut

class GameCreate(BaseModel):
    opponent: str
    date: datetime
    time: Optional[str] = None
    location: Optional[str] = None
    category: Optional[str] = None
    status: Optional[str] = None
    owner_id: Optional[int] = None
    # Adicione outros campos necessários para criação

class GameUpdate(BaseModel):
    opponent: Optional[str] = None
    date: Optional[datetime] = None
    time: Optional[str] = None
    location: Optional[str] = None
    category: Optional[str] = None
    status: Optional[str] = None
    owner_id: Optional[int] = None
    players: Optional[List[int]] = None
    # Adicione outros campos necessários para atualização

class GameOut(BaseModel):
    id: int
    opponent: str
    date: datetime
    time: Optional[str] = None
    location: Optional[str] = None
    category: Optional[str] = None
    status: str
    owner_id: Optional[int] = None
    created_at: datetime
    players: List[PlayerOut] = []
    # Adicione outros campos/relacionamentos se necessário

    class Config:
        from_attributes = True 